Weekly CI で muon-master job が失敗した

Weekly CI で muon-master job が失敗した

<!-- SPDX-License-Identifier: FSFAP -->
<!--
バグの報告ありがとうございます！
以下の項目は修正の手掛かり・助けになりますので記入をお願いいたします。
不具合なのかはっきりしない動作や挙動はDiscussionsで質問してみてください。
-->

**バグの説明**
<!-- 何が起こったか、エラーメッセージがあればそれも書く -->

GitHub Actions の Weekly CI で muon-master job が失敗したと通知が届きました。

https://github.com/JDimproved/JDim/actions/runs/11656324308/job/32452280013

muon-master job の raw logs (抜粋):
```
2024-11-04T01:05:15.7245849Z ##[group]Run git clone --depth 1 https://git.sr.ht/~lattis/muon muon-src
2024-11-04T01:05:15.7246635Z [36;1mgit clone --depth 1 https://git.sr.ht/~lattis/muon muon-src[0m
2024-11-04T01:05:15.7304900Z shell: /usr/bin/bash -e {0}
2024-11-04T01:05:15.7305325Z env:
2024-11-04T01:05:15.7305702Z   CC: gcc-11
2024-11-04T01:05:15.7306034Z   CXX: g++-11
2024-11-04T01:05:15.7306369Z ##[endgroup]
2024-11-04T01:05:15.7394983Z Cloning into 'muon-src'...
2024-11-04T01:05:17.4050627Z ##[group]Run git -C muon-src log -n1
2024-11-04T01:05:17.4051118Z [36;1mgit -C muon-src log -n1[0m
2024-11-04T01:05:17.4108454Z shell: /usr/bin/bash -e {0}
2024-11-04T01:05:17.4108937Z env:
2024-11-04T01:05:17.4109223Z   CC: gcc-11
2024-11-04T01:05:17.4109537Z   CXX: g++-11
2024-11-04T01:05:17.4109938Z ##[endgroup]
2024-11-04T01:05:17.4200515Z commit 2c90b85892ea72f3120ee73f58de3eaa4d2eec2f
2024-11-04T01:05:17.4201303Z Author: Stone Tickle <lattis@mochiro.moe>
2024-11-04T01:05:17.4202284Z Date:   Sun Nov 3 13:10:15 2024 -0500
2024-11-04T01:05:17.4203663Z 
2024-11-04T01:05:17.4204485Z     windows run_cmd add wait timeout to ioport polling
2024-11-04T01:05:17.4233123Z ##[group]Run cd muon-src
2024-11-04T01:05:17.4233652Z [36;1mcd muon-src[0m
2024-11-04T01:05:17.4234038Z [36;1m./bootstrap.sh stage1[0m
2024-11-04T01:05:17.4234399Z [36;1m./stage1/muon version[0m
2024-11-04T01:05:17.4234859Z [36;1mcd ..[0m
2024-11-04T01:05:17.4285393Z shell: /usr/bin/bash -e {0}
2024-11-04T01:05:17.4285887Z env:
2024-11-04T01:05:17.4286151Z   CC: gcc-11
2024-11-04T01:05:17.4286483Z   CXX: g++-11
2024-11-04T01:05:17.4286863Z ##[endgroup]
2024-11-04T01:05:17.4366880Z + dir=stage1
2024-11-04T01:05:17.4367461Z + mkdir -p stage1
2024-11-04T01:05:17.4378537Z + pkgconf_cmd=
2024-11-04T01:05:17.4379282Z + command -v pkgconf
2024-11-04T01:05:17.4379787Z + pkgconf_cmd=pkgconf
2024-11-04T01:05:17.4380264Z + [ -n pkgconf ]
2024-11-04T01:05:17.4380696Z + pkgconf libpkgconf
2024-11-04T01:05:17.4397088Z + pkgconf --cflags libpkgconf
2024-11-04T01:05:17.4409258Z + pkgconf_cflags=-I/usr/include/pkgconf  -DBOOTSTRAP_HAVE_LIBPKGCONF
2024-11-04T01:05:17.4410430Z + pkgconf --libs libpkgconf
2024-11-04T01:05:17.4420608Z + pkgconf_libs=-lpkgconf 
2024-11-04T01:05:17.4421972Z + gcc-11 -Iinclude -I/usr/include/pkgconf -DBOOTSTRAP_HAVE_LIBPKGCONF src/amalgam.c -lpkgconf -o stage1/muon-bootstrap
2024-11-04T01:05:19.7325363Z /home/runner/work/_temp/f86bb109-09d0-406a-9166-934f50d51366.sh: line 3: ./stage1/muon: No such file or directory
2024-11-04T01:05:19.7340835Z ##[error]Process completed with exit code 127.
```

**再現の方法**
<!-- バグの動作を再現する手順を書く -->
```sh
git clone --depth 1 https://git.sr.ht/~lattis/muon muon-src
cd muon-src
./bootstrap.sh stage1
./stage1/muon version
```

**やりたかったこと・期待する結果**
<!-- 本来の望ましい動作を書く -->
job を実行して成功すること。

**スクリーンショット**
<!-- 表示の問題はスクリーンショットがあれば分かりやすい -->
なし

**動作環境**
<!-- クリップボードへのコピーを利用する -->

JDim:
```
[バージョン] JDim 0.13.0-alpha20241026(git:52bc133a1b)
[ディストリ ] Ubuntu 24.10 (x86_64)
[パッケージ] バイナリ/ソース( <配布元> )
[ DE／WM ] KDE (Wayland)
[　gtkmm 　] 3.24.9
[　glibmm 　] 2.66.7
[　TLS lib　] GnuTLS 3.8.6
[オプション ] '--with-alsa''--with-pangolayout'
[ そ の 他 ]
```

muon:
```
commit 2c90b85892ea72f3120ee73f58de3eaa4d2eec2f (HEAD -> master, origin/master, origin/HEAD)
Author: Stone Tickle <lattis@mochiro.moe>
Date:   2024-11-03 13:10:15 -0500

    windows run_cmd add wait timeout to ioport polling
```

**追加の情報**
<!-- ユーザーエージェントなどが手掛かりとなる情報 -->

muon リポジトリに対して git bisect で原因のコミットを探索した結果です。
```
6ec469bb42b28f3a1943d58c0e15a8493220c2f7 is the first bad commit
commit 6ec469bb42b28f3a1943d58c0e15a8493220c2f7
Author: Stone Tickle <lattis@mochiro.moe>
Date:   2024-11-03 07:27:25 -0500

    bootstrap.sh - use muon-bootstrap as exe name

    This makes the behavior the same as windows and ensures that the build
    process doesn't have spurious errors due to trying to use the muon exe
    while it is being written to.

 .github/workflows/macos.yml  | 8 ++++----
 .github/workflows/msys2.yml  | 2 +-
 .github/workflows/ubuntu.yml | 8 ++++----
 README.md                    | 8 ++++----
 bootstrap.sh                 | 2 +-
 tools/ci/bootstrap.sh        | 4 ++--
 tools/ci/solaris11.sh        | 6 +++---
 7 files changed, 19 insertions(+), 19 deletions(-)
bisect found first bad commit
```

bootstrapping (bootstrap.sh) で生成した実行ファイルの名前が muon-bootstrap に変更されたようです。

--- a/.github/workflows/weekly.yml
+++ b/.github/workflows/weekly.yml
@@ -83,20 +83,20 @@ jobs:
       - name: git clone muon
         run: |
           git clone --depth 1 https://git.sr.ht/~lattis/muon muon-src
-        # Skip bootstrapping muon to do meson setup instead.
       - name: git log -n1 muon
         run: git -C muon-src log -n1
+        # Since 2024-11-03 07:27:25 -0500 (6ec469bb42), bootstrapping has used muon-bootstrap as exe name.
       - name: build muon stage1
         run: |
           cd muon-src
           ./bootstrap.sh stage1
-          ./stage1/muon version
+          ./stage1/muon-bootstrap version
           cd ..
       - name: build muon stage2
         run: |
           cd muon-src
-          ./stage1/muon setup -Ddocs=disabled -Dlibarchive=disabled -Dlibcurl=disabled -Dlibpkgconf=enabled ../muon-build
-          ./stage1/muon samu -C ../muon-build
+          ./stage1/muon-bootstrap setup -Ddocs=disabled -Dlibarchive=disabled -Dlibcurl=disabled -Dlibpkgconf=enabled ../muon-build
+          ./stage1/muon-bootstrap samu -C ../muon-build
           cd ..
       - name: muon version
         run: ./muon-build/muon version

